code/modules/library/lib_items.dm

```diff
@@ -96,6 +96,56 @@
 		icon_state = "book-5"
 
 
+/obj/structure/bookcase/libraryspawn
+	var/spawn_category
+	var/spawn_amount = 3
+
+/obj/structure/bookcase/libraryspawn/Initialize()
+
+	. = ..()
+
+	name = "[initial(name)] ([spawn_category])"
+
+	if(!config.sql_enabled)
+		return
+
+	establish_db_connection(dbcon)
+
+	if(!dbcon.IsConnected())
+		log_debug(" Library Book Spawning: SQL ERROR - Failed to connect and spawn books for [spawn_category].")

```

Comment:
whitespace at the start of log_debug

Reply:
the worst mistake a coder could make

rescinding my dev application now